packages/core/src/internal/base/button.base.ts

```diff
@@ -27,189 +37,5 @@ export class CdsBaseButton extends LitElement {
-   */
-  private triggerNativeButtonBehavior(event: Event) {
-    if (!this.readonly) {
-      if (this.disabled) {
-        stopEvent(event);
-      } else if (!event.defaultPrevented && this.marker && this.explicitClick) {
-        if (!this.markerAttached) {
-          this.appendChild(this.marker);
-        }
-        this.querySelector('button[aria-hidden]')?.dispatchEvent(
-          new MouseEvent('click', { relatedTarget: this, composed: true })
-        );
-        if (this.type !== 'submit') {
-          this.marker.remove();
-        }
-      }
-    }
-    // get ready for the next real click check
-    this.explicitClick = false;
-  }
-
-  private createMarker() {
-    const buttonTemplate = html`<button
-      aria-hidden="true"
-      role="presentation"
-      ?disabled="${this.disabled}"
-      tabindex="-1"
-      style="display: none !important"
-      value="${ifDefined(this.value)}"
-      name="${ifDefined(this.name)}"
-      type="${ifDefined(this.type)}"
-    ></button>`;
-
-    const marker = document.createElement('div');
-    // avoid an empty space inside button
-    marker.style.display = 'none';
-    render(buttonTemplate, marker);
-    this.marker = marker;
-  }
-
-  private get markerAttached(): boolean {
-    return !!(this.marker && this.marker.parentElement);
-  }
-
-  // A marker significant property is one that should be dynamically bound to a property of the native button
-  private hasMarkerSignificantProperty(props: Map<string, any>): boolean {
-    for (const prop of this.markerSignificantProperties) {
-      if (props.has(prop)) {
-        return true;
-      }
-    }
-    return false;
-  }
-
-  private emulateKeyBoardEventBehavior(evt: KeyboardEvent) {
-    onAnyKey(['enter', 'space'], evt, () => {
-      if (!this.readonly) {
-        this.click();
-        stopEvent(evt);
-      }
-    });
-  }
-
-  private updateButtonAttributes() {
-    this.isAnchor = this.parentElement?.tagName === 'A';
-
-    if (this.isAnchor && this.parentElement) {
-      this.parentElement.style.lineHeight = '0';
-      this.parentElement.style.textDecoration = 'none'; // fixes issue when style is applied to text node
-    }
-
-    this.readonly = this.readonly || this.isAnchor;
-
-    if (this.readonly) {
-      this.role = null;
-      this.tabIndexAttr = null;
-      this.ariaDisabled = null;
-    } else {
-      this.role = 'button';
-      this.tabIndexAttr = this.disabled ? -1 : 0;
-      this.ariaDisabled = this.disabled ? 'true' : 'false';
-    }
-  }
-
-  // Forms handle default submit buttons without type="submit" automatically.
-  // Because of the dynamic createion of the native element, we prefer to set the attribute explicitly.
-  private addDefaultSubmitBehavior() {
-    if (!this.type && this.closest('form')) {
-      this.type = 'submit';
-    }
-    if (this.type === 'submit' && !this.markerAttached) {
-      this.appendChild(this.marker);
-    }
-  }
```

Comment:
all logic is split into the controllers defined above 

```typescript
@active<CdsBaseButton>()
@ariaButton<CdsBaseButton>()
@ariaPressed<CdsBaseButton>()
@ariaDisabled<CdsBaseButton>()
@ariaPopupTrigger<CdsBaseButton>()
@buttonSubmit<CdsBaseButton>()
@buttonAnchor<CdsBaseButton>()
```

This makes the logic generic and reusable for other components that may need similar a11y behaviors but are not a traditional button. This also helps ensure our APIs are consistent with how components state is described (disabled, expanded, pressed, readonly...)